### ISSUE TITLE
Why I set env: { node: false }, but usage of __filename is still allowed?

### ISSUE THREAD

**Issue by lovetingyuan** [eslint/eslint#14940]
**Tell us about your environment**

* **ESLint Version:** v7.32.0
* **Node Version:** v16.4.0
* **npm Version:** 7.18.1
* **Operating System:** mac os

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
'vue2': {
      env: {
        browser: true,
        es2020: true,
        node: false,
      },
      parserOptions: {
        ecmaVersion: 2020,
        ecmaFeatures: {
          jsx: true,
          legacyDecorators: false
        },
      },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash

```

**What did you expect to happen?**

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

**Steps to reproduce this issue:**

1. 
2. 
3. 

**Are you willing to submit a pull request to fix this bug?**
